lib/beaker-puppet/install_utils/puppet_utils.rb

```diff
@@ -90,41 +90,51 @@ def get_puppet_collection(agent_version = 'latest')
           collection
         end
 
-        # Determine the puppet collection that matches a given version of the puppet-agent
-        # package (you can find this version in the `aio_agent_version` fact).
+        # Determine the puppet collection that matches a given package version. The package
+        # must be one of
+        #   * :puppet_agent (you can get this version from the `aio_agent_version_fact`)
+        #   * :puppet
         #
-        # @param agent_version [String] a semver version number of the puppet-agent package, or the string 'latest'
+        #
+        # @param package [Symbol] the package name. must be one of :puppet_agent, :puppet
+        # @param version [String] a version number or the string 'latest'
         # @returns [String|nil] the name of the corresponding puppet collection, if any
-        def puppet_collection_for_puppet_agent_version(agent_version)
-          agent_version = agent_version.to_s
-          return 'puppet' if agent_version.strip == 'latest'
+        def puppet_collection_for(package, version)
+          valid_packages = [
+            :puppet_agent,
+            :puppet
```

Comment:
Shouldn't we have a longer list of valid packages here? Puppetserver, puppetdb, puppetdb-termini are all versioned on the same scheme as puppet and puppet_agent, so there is a case to be made that we might need to find the release stream for those packages.

http://apt.puppet.com/pool/jessie/puppet6/p/

Reply:
I was thinking this could be something that'd be extended on a per-need case? So if a puppetserver person finds out that they need to calculate the collection from their package version, then they can extend this method with that code (and they'd figure out they need the collection from the raised exception). If you think it is a good idea to do as much as we can in the current implementation, however, I'm happy to update it. I'm not sure what the collection should be for legacy server/puppetdb versions (like Server 2.x, since agent 1.x uses it). Maybe we don't need to worry about those.

Reply:
I suppose it depends on where this method is used? But generally since this repo is so ad hoc, I'm fine waiting to add those other packages until we need them. As long as that's a conscious decision